### ISSUE TITLE
Cruel Reality issue

### ISSUE THREAD

**Issue by ChickenSquad** [magefree/mage#3191]
The new enchantment cruel reality is giving the opponents the option to not to sacrifice a creature a simply opt to lose the 5 life. The card should not give the opponent the option to sacrifice. 

The card reads-

At the beginning of enchanted player's upkeep, that player sacrifices a creature or planeswalker. If the player can't, he or she loses 5 life.

**Comment by ingmargoudt** [comment]
just checked the implementation of Cruel Reality and indeed there seems to be an optional sacrifice.

**Comment by drmDev** [comment]
Looking at similar cards, probably missing something similar to:

```
target.canChoose(source.getSourceId(), player.getId(), game)) {
            while (!target.isChosen() && target.canChoose(player.getId(), game) && player.canRespond()) {
                player.chooseTarget(Outcome.Sacrifice, target, source, game);
            }
```

**Comment by stravant** [comment]
apply() also currently returns false when the life payment is made instead of sacrificing a creature, it should return true in that case.

**Comment by jeffwadsworth** [comment]
@stravant That is correct, the loss of 5 life should have a return true below it.  As for the other points, there is no option not to sacrifice if a creature or planeswalker is present.

**Comment by magenoxx** [comment]
one question: how will it interact with Sigarda, Host of Herons?
Flying, hexproof
Spells and abilities your opponents control can't cause you to sacrifice permanents.

**Comment by spjspj** [comment]
You should lose the 5 life if you control Sigarda, Host of Herons.

**Comment by jeffwadsworth** [comment]
@magenoxx I will test that scenario.

**Comment by drmDev** [comment]
" As for the other points, there is no option not to sacrifice if a creature or planeswalker is present." to me it looked like there was? It looked like it was something like give the player the option to choose with creature or planeswalker, and then do some stuff if that occurred, otherwise they lose 5 life. So players were opting not to select a creature/pw and lost 5 life instead.

Another card where unit tests on a few scenarios would be helpful... le sigh I'll create them later tonight if no one else will

**Comment by jeffwadsworth** [comment]
@drmDev Play the card in the game and see if you can find an optional way out of paying the sacrifice.  I could not.

**Comment by jeffwadsworth** [comment]
@magenoxx It appears that Sigarda, Host of Herons does not check for StackAbility.  I added a fix for that.

**Comment by drmDev** [comment]
Tested all the basic scenarios I could think of, cannot duplicate the issue. Forgot to add a Sigarda test oh well, another time.

**Comment by fireshoes** [comment]
I played someone today that I enchanted with Cruel Reality. The trigger to sacrifice occurred and they lost 5 life instead of saccing one of their two creatures. He said there was a yes/no prompt and he clicked no to see what would happen.

**Comment by fireshoes** [comment]
Cruel Reality trigger. Opp loses 5 life. Casts Crocodile of the Crossing and attacks with it _and Greater Sandwurm_.

`<font color="#CCCC33">11:50 PM: </font><font color="White">Ability triggers: </font><font color="#696969"><a href="#Cruel Reality">Cruel 
    Reality</a></font><font color="White"> [b86] - At the beginning of 
    enchanted player's upkeep, that player sacrifices a creature or 
    planeswalker. If the player can't, he or she loses 5 life.</font>
<font color="#CCCC33">11:50 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    loses 5 life</font>
<font color="#CCCC33">11:50 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    draws a card</font>
<font color="#CCCC33">11:51 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    puts </font><font color="#B0C4DE"><a href="#Swamp">Swamp</a></font><font color="White"> 
    [500] from hand onto the Battlefield</font>
<font color="#CCCC33">11:51 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    plays </font><font color="#B0C4DE"><a href="#Swamp">Swamp</a></font><font color="White"> 
    [500]</font>
<font color="#CCCC33">11:51 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    casts </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7]</font>
<font color="#CCCC33">11:51 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    puts </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7] from stack onto the 
    Battlefield</font>
<font color="#CCCC33">11:51 PM: </font><font color="White">Ability triggers: </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7] - When </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7] enters the 
    battlefield, put a -1/-1 counter on target creature you control. - 
    targeting </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7] </font>
<font color="#CCCC33">11:51 PM: </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7]: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    puts 1 -1/-1 counter on </font><font color="#90EE90"><a href="#Crocodile of the Crossing">Crocodile 
    of the Crossing</a></font><font color="White"> [af7]</font>
<font color="#CCCC33">11:51 PM: </font><font color="#20B2AA">TripleD</font><font color="White"> 
    attacks with 2 creatures</font>
<font color="#CCCC33">11:52 PM: </font><font color="White">Attacker: </font><font color="#90EE90"><a href="#Greater Sandwurm">Greater 
    Sandwurm</a></font><font color="White"> [b5d] (7/7) blocked by </font><font color="#696969"><a href="#Cursed Minotaur">Cursed 
    Minotaur</a></font><font color="White"> [8e5] (3/2) </font>`

**Comment by jeffwadsworth** [comment]
Looks like he is living in the twilight zone.  This is using the latest client?

**Comment by drmDev** [comment]
I even added tests for trying to setChoice to "No"
```
+        String cReality = "Cruel Reality";
+        String memnite = "Memnite"; // {0} 1/1
+
+        addCard(Zone.HAND, playerA, cReality);
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 7);
+        addCard(Zone.BATTLEFIELD, playerB, memnite);
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, cReality, playerB);
+        setChoice(playerB, "No");
+
+        setStopAt(2, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        assertGraveyardCount(playerB, memnite, 1);
+        assertPermanentCount(playerA, cReality, 1);
+        assertLife(playerB, 20);
```
see commit 3246bd4551f8912e674aacebededecb857182ca0

**Comment by drmDev** [comment]
I don't have a good explanation for it still occurring - not reproducible at the moment my manual or automated tests from the latest builds on master. perhaps some framework change affected this card working correctly between release and now? or the card was patched between then and now?

**Comment by fireshoes** [comment]
I was on dev client. Don't know on opp.

**Comment by jeffwadsworth** [comment]
He must be using the old client.  It didn't have the code change yet.  It will be fine once the new version is released.

**Comment by ChickenSquad** [comment]
Thank you for your stellar work! You guys are simply the best!
